prober/prober.go

```diff
@@ -127,12 +140,26 @@ func (p *Probe) Close() error {
 }
 
 // probeLoop invokes runProbe on fun every interval. The first probe
-// is run after interval.
+// is run after a random delay (if spreading is enabled) or immediately.
 func (p *Probe) loop() {
 	defer close(p.stopped)
 
-	// Do a first probe right away, so that the prober immediately exports results for everything.
-	p.run()
+	if p.prober.spread && p.initialDelay > 0 {
```

Comment:
We certainly can make it random. However it occurs to me that the caller knows how many probers it is planning to add, and the interval it expects to have them probe. The caller could possibly pass in a value to spread the probers evenly across the interval.

Reply:
That's a good point. Currently the API to add a probe ([Run](https://github.com/tailscale/tailscale/blob/adec726fee58a8d5c8e54fed789abced23737cdb/prober/prober.go#L60)) is pretty hard to add optional parameters to. I am planning to change that to enable custom metrics reported by probes, and as part of that change I can also add ability for an individual probe to set its own `initialDelay`.

Reply:
SGTM